tests/core/daemon/test_daemon_register.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import annotations
+
+import aiohttp
+import pytest
+
+from chia.daemon.server import WebSocketServer
+from chia.simulator.block_tools import BlockTools
+from chia.util.ws_message import create_payload
+
+
+@pytest.mark.asyncio
+async def test_multiple_register_same(get_daemon: WebSocketServer, bt: BlockTools) -> None:
+    ws_server = get_daemon
+    config = bt.config
+
+    daemon_port = config["daemon_port"]
+
+    # setup receive service to connect to the daemon
+    async with aiohttp.ClientSession() as client:
+        ws = await client.ws_connect(
+            f"wss://127.0.0.1:{daemon_port}",
+            autoclose=True,
+            autoping=True,
+            ssl_context=bt.get_daemon_ssl_context(),
+            max_msg_size=100 * 1024 * 1024,
+        )
+
+        service_name = "test_service"
+        data = {"service": service_name}
+        payload = create_payload("register_service", data, service_name, "daemon")
+        for _ in range(4):
+            await ws.send_str(payload)
+            await ws.receive()
+
+    connections = ws_server.connections.get(service_name, set())
+    assert len(connections) == 1
```

Comment:
How about verifying 0 prior to registering, and then 1 after each registration in the loop?  Or collect the lengths in a list and `assert that_list = [0, 1, 1, 1, 1]` to get them all asserted at once at the end.